color_refinement.py

```diff
@@ -78,6 +79,60 @@ def color_refine(coloring: "Coloring") -> "Coloring":
     return coloring
 
 
+def fast_color_refine(graph: "Graph"):
+    # takes initial coloring
+    qlist = DoubleLinkedList()
+    color_map = {}
+    neigh_map = {}
+    #TODO: Replace this with better things, because it's now just used to get the amount of colors
+    # because I did not figure out how to do this with the Coloring class yet
+    # Neigh_map might not be neccessary
+    for v in graph.vertices:
+        if v.colornum not in color_map.keys():
+            color_map[v.colornum] = []
+        color_map[v.colornum].append(v)
+        if v not in neigh_map.keys():
+            neigh_map[v] = None
+        neigh_map[v] = v.neighbours
+    first_color = sorted(list(color_map.keys()))[0]
+    qlist.append(Node(first_color))
```

Comment:
Moeten volgens het algoritme niet alle kleuren in de eerste iteratie worden toegevoegd?

Reply:
Volgens mij stond alleen de eerste kleur in de queue (maar het kan zijn dat ik dat verkeerd begrepen heb